### ISSUE TITLE
No support for features in DeviceDescriptor

### ISSUE THREAD

**Issue by JeffreyPalmer** [JuliaWGPU/WGPUCore.jl#23]
Hi!

I was attempting to use some of the WebGPU features (specifically `WGPUFeatureName_Float32Filterable`) when I noticed that there is currently no support for any additional feature requests in the device implementation. Is that something that you have added elsewhere, or should I put together a PR to add support for them?

Thanks!

**Comment by arhik** [comment]
@JeffreyPalmer Currently we lack basic api to query features in `adapter.jl` file, choose device, and request feature in `device.jl`. 

Choosing device is a separate issue here https://github.com/JuliaWGPU/WGPUgfx.jl/issues/29.  (Should have been raised in `WGPUCore`)

Our approach so far has been need-basis. We are filling API as we need them. We would definitely want to have this and I will add that to the feature tracking list here https://github.com/orgs/JuliaWGPU/projects/1. If you have bandwidth to work on it PR would be great.

**Comment by JeffreyPalmer** [comment]
Thanks for the feedback. I'm relatively new to Julia FFI stuff, but I think I can probably solve my immediate need and go from there. I'll take a look. 🙏

**Comment by JeffreyPalmer** [comment]
Sorry I ghosted this conversation. I ended up using OpenGL for my immediate needs, due to the stability of the package, but I'm still very interested in this work as it moves forward. 🙏

Please feel free to close this issue if you'd like.

**Comment by arhik** [comment]
There was 3-5 months gap in our contributions to JuliaWGPU (I was busy with my startup). Since we are lagging with upstream WGPU version, We will start working on WGPUNative updates soon anyways. It will then be ripe time to address this issue too. Until then lets keep it open. Hopefully it can be addressed in 1-2 weeks. Apologies for keeping you wait on it.
